src/Domain/NotificationPoll.php

```diff
@@ -20,15 +20,15 @@ public static function fromArray(array $json): NotificationPoll
     {
         return new NotificationPoll(
             $json['count'],
-            $json['notification'] ?? ''
+            isset($json['notification']) ? Notification::fromArray($json['notification']) : null
```

Comment:
It's almost perfect, the only thing I would like to change is the use of the `isset()` function. In the rest of the codebase the following format is used:
```php
$json['notification'] ? Notification::fromArray($json['notification']) : null
```
It would be nice if you could update this for the sake of consistency. After that, I will gladly merge this 👍

Reply:
`src/Domain/DomainDetails.php` is using this format. Which is consistent for this type of optional variable data that utilizes a class.

```php
isset($json['zone']) ? Zone::fromArray($json['zone']) : null,
isset($json['contacts']) ? DomainContactCollection::fromArray($json['contacts']) : null,
isset($json['keyData']) ? KeyDataCollection::fromArray($json['keyData']) : null,
isset($json['ds_data']) ? DsDataCollection::fromArray($json['ds_data']) : null
```

---

If I remove `isset()` from the equation I get a notice of undefined `notification` key in `$json['notification']`.

```php
$json['notification'] ? Notification::fromArray($json['notification']) : null
```

The same failed test result happens when I remove the `isset()` from `DomainDetails` as well.

```php
$json['zone'] ? Zone::fromArray($json['zone']) : null,
```

---

I assume this is because E_NOTICE is turned off during development.